google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeploymentConfigurator.java

```diff
@@ -112,9 +69,20 @@ public AppEngineDeploymentConfiguration createDefaultConfiguration(
   public SettingsEditor<AppEngineDeploymentConfiguration> createEditor(
       @NotNull DeploymentSource source,
       @NotNull RemoteServer<AppEngineServerConfiguration> server) {
+    if (!(source instanceof AppEngineDeployable)) {
+      logger.error(
+          String.format("Deployment source with name %s is not deployable to App Engine.",
+              source.getPresentableName()));
+      return null;
+    }
+
     return new AppEngineDeploymentRunConfigurationEditor(
         project,
         source,
-        new CloudSdkAppEngineHelper(new File(server.getConfiguration().getCloudSdkHomePath())));
+        ((AppEngineDeployable) source).getEnvironment(),
```

Comment:
if source has the environment and is already a parameter, why does the client need to redundantly pass it here?